### PR TITLE
Make JsonSchemaAssertions check for JSON content-type

### LINKED ISSUE #59: Make JsonSchemaAssertions check for JSON content-type
The `JsonSchemaAssertions` object does not assert for `application/json` in the content-type header. It should behave similar to `JsonResponseAssertion` in that it checks that the content-type is `application/json`. It should also allow the user to override the expected content-type using `overrideContentType()`.

### PR DESCRIPTION
**Pull request by maria-camenzuli**
Added assertions on the response content type in `JsonSchemaAssertions`.

This fixes #59

### CODE REVIEW

**Comment by maria-camenzuli** [comment]
This change was simply copying the assertions on the content type from `JsonResponseAssertions` over to `JsonSchemaAssertions`. Can we come up with some kind of abstraction for the capabilities of the assertion classes to avoid this duplication? @FrelliBB @KPull

**Comment by KPull** [comment]
> Can we come up with some kind of abstraction for the capabilities of the assertion classes to avoid this duplication?

Yes. I can forsee that an `Assertions` object would typically want to assert the status code and content-type for example. We can abstract this into a `CommonAssertions` object (just like we have `CommonRequestAttributes`), and have the applicable assertions inherit from `CommonAssertions` by composition.

File: src/test/java/rocks/bastion/core/assertions/TestModelResponse.java [hunk 1]
```diff
@@ -20,4 +20,19 @@
...
 
+    /**
+     * Prepare a model response object with a status code of HTTP 200, as well as the given JSON body and content type,
+     * for testing purposes.
+     *
+     * @param jsonContent
+     * @param contentType
+     * @return A model response object.
+     */
+    static ModelResponse<String> prepare(String jsonContent, String contentType) {
```
Comment (KPull) [inline src/test/java/rocks/bastion/core/assertions/TestModelResponse.java]: This method appears to be a generalisation of the method above it. I would rewrite (and rename) the method above this one to use the new method you wrote.

**Comment by maria-camenzuli** [comment]
I think it's best if we refactor the structure of the assertions in a separate pull request. I can open an issue.

**Comment by maria-camenzuli** [comment]
I've applied the other changes requested.

**Comment by maria-camenzuli** [comment]
Opened #66 for the assertion abstraction task.

**Pull request merged**